packages/perseus/src/widgets/table/utils.ts

```diff
@@ -0,0 +1,14 @@
+import type {PerseusTableUserInput} from "../../validation.types";
+
+/**
+ * Filters the given table (modelled as a 2D array) to remove any rows that are
+ * completely empty.
+ *
+ * @returns A new table with only non-empty rows.
+ */
+export const filterNonEmpty = function (table: PerseusTableUserInput) {
```

Comment:
I noticed the input parameter here is `table`, not `userInput`. I think that was probably on purpose since this is a util function, but pointing it out just in case.

Reply:
Good eye @Myranae.  In this case I named it "table" because we call it with both the use input but also the rubric's solution data. What I think should change is the type here to make it clearer that we aren't only accepting user input. 

<img width="433" alt="image" src="https://github.com/user-attachments/assets/603c4033-dab9-413e-b259-5e52a3dd215f">